crates/xline/src/utils/trace.rs

```diff
@@ -1,9 +1,24 @@
-use anyhow::Result;
+use anyhow::{Ok, Result};
 use opentelemetry_contrib::trace::exporter::jaeger_json::JaegerJsonExporter;
 use opentelemetry_sdk::runtime::Tokio;
+use tracing::warn;
 use tracing_appender::non_blocking::WorkerGuard;
 use tracing_subscriber::{fmt::format, layer::SubscriberExt, util::SubscriberInitExt, Layer};
-use utils::config::{file_appender, LogConfig, TraceConfig};
+use utils::config::{default_rotation, file_appender, LogConfig, TraceConfig};
+
+/// Return a Box trait from the config
+fn generate_writer(name: &str, log_config: &LogConfig) -> Box<dyn std::io::Write + Send> {
+    match *log_config.path() {
+        Some(ref file_path) if file_path.to_string_lossy() == "/stdout" => {
+            if *log_config.rotation() != default_rotation() {
+                warn!("The log is output to the terminal, so the rotation parameter is ignored.");
+            }
+            Box::new(std::io::stdout())
+        }
+        Some(ref file_path) => Box::new(file_appender(*log_config.rotation(), file_path, name)),
+        None => unreachable!(),
```

Comment:
Please add some information in this `unreachable` macro.

Reply:
Got it